Remove first party uses of distutils

jgraham opened the pull request.

jgraham committed:
Remove vendored enum

This is no longer required for Python 3

jgraham committed:
Replace distutils.spawn.find_executable with shutil.which

jgraham committed:
Replace distutils.version.LooseVersion with packaging.version.Version

gsnedders commented on tools/wptrunner/wptrunner/browsers/safari.py:
```diff
@@ -2,8 +2,8 @@
 
 import os
 import plistlib
+from packaging.version import Version
```
Can we make `packaging` an explicit requirement if we're going to start depending on it directly?

jgraham committed:
Make packaging an explicit requirement

We could make this a requirement just for Safari and WKTR, since they're the only browsers
that really use it, but we're getting it pulled in anyway, so this is probably fine.

gsnedders reviewed (approved):
Would also be good to update expectations for infrastructure, but this looks good to me
